File: java/src/test/java/com/testexercise/ebayframework/AutomationFramework.java
package com.testexercise.ebayframework;

import java.net.*;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.*;

public class AutomationFramework
{
    private final WebDriver driver;
    private final String baseUri;
    
    public String baseUri()
    {
        return baseUri;
    }
    
    public AutomationFramework(WebDriver driver, String baseUri)
            throws URISyntaxException
    {
        if (driver == null)
            throw new IllegalArgumentException("driver cannot be null");
        if (baseUri == null)
            throw new IllegalArgumentException("baseUri cannot be null");
        
        this.driver = driver;
        this.baseUri = baseUri;
        init();
    }
    
    private void init()
    {
        driver.manage().
            timeouts().
            implicitlyWait(1000, TimeUnit.MILLISECONDS);
    }
    
    public WebDriver driver()
    {
        return driver;
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/AutomationFramework.cs
using System;
using OpenQA.Selenium;

namespace TestExercise.EbayPOMFramework
{
    public class AutomationFramework
    {
        public string BaseUri { get; }
        public IWebDriver Driver { get; }

        public AutomationFramework(IWebDriver driver, String baseUri)
        {
            if (driver == null)
                throw new ArgumentNullException("driver cannot be null");
            if (baseUri == null)
                throw new ArgumentNullException("baseUri cannot be null");

            this.Driver = driver;
            this.BaseUri = baseUri;
            Init();
        }

        private void Init()
        {
            Driver.Manage().Timeouts().ImplicitWait = new TimeSpan(0, 0, 1);
        }
    }
}